File: src/main/java/app/gamestate/HelpState.java
package app.gamestate;

import app.manager.*;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.paint.Color;
import javafx.scene.text.Font;
import javafx.scene.text.TextAlignment;

public class HelpState extends GameStateBase {

    private final String[] labelsControls = {
            LanguageManager.INSTANCE.get("help_attack"),
            LanguageManager.INSTANCE.get("help_run"),
            LanguageManager.INSTANCE.get("help_pause"),
            LanguageManager.INSTANCE.get("help_move"),
    };
    private final String labelExit = LanguageManager.INSTANCE.get("help_exit");
    private final String labelHelp = LanguageManager.INSTANCE.get("menu_help");

    private Font pusab40;
    private Font pusab50;

    public HelpState(GameStateManager gameStateManager) {
        super(gameStateManager);
        initialize();
    }

    @Override
    public void initialize() {
        pusab40 = TextManager.loadFont(TextManager.PUSAB, 40);
        pusab50 = TextManager.loadFont(TextManager.PUSAB, 50);
    }

    @Override
    public void update() {
        handleInput();
    }

    @Override
    public void draw(GraphicsContext graphics) {
        graphics.drawImage(TextureManager.INSTANCE.getImage("state-help"), 0, 0);
        graphics.setFill(Color.BLACK);

        graphics.setTextAlign(TextAlignment.CENTER);
        graphics.setFont(pusab50);
        graphics.fillText(labelHelp, 320, 120);

        graphics.setFont(pusab40);
        graphics.setTextAlign(TextAlignment.LEFT);
        graphics.fillText(labelsControls[0], 330, 210);
        graphics.fillText(labelsControls[1], 330, 280);
        graphics.fillText(labelsControls[2], 330, 355);
        graphics.fillText(labelsControls[3], 330, 460);

        graphics.setFont(pusab50);
        graphics.setFill(Color.RED);
        graphics.setTextAlign(TextAlignment.CENTER);
        graphics.fillText(labelExit, 320, 545);
        graphics.drawImage(TextureManager.INSTANCE.getImage("arrow-right"), 180, 512);
    }

    @Override
    public void handleInput() {
        if (KeyManager.isPressed(KeyManager.K_ENTER)) {
            getGameStateManager().popState();
        }
    }
}

File: src/main/kotlin/app/gamestate/HelpState.kt
package app.gamestate

import app.manager.GameStateManager
import app.manager.KeyManager
import app.manager.LanguageManager.get
import app.manager.TextureManager.getImage
import app.utils.PUSAB
import app.utils.loadFont
import javafx.scene.canvas.GraphicsContext
import javafx.scene.paint.Color
import javafx.scene.text.TextAlignment
import org.apache.logging.log4j.LogManager

private val logger = LogManager.getLogger(HelpState::class.java)
class HelpState(gameStateManager: GameStateManager) : GameStateBase(gameStateManager) {
    private val labelsControls = listOf(
            get("help_attack"),
            get("help_run"),
            get("help_pause"),
            get("help_move"))
    private val labelExit = get("help_exit")
    private val labelHelp = get("menu_help")

    private val pusab40 = loadFont(PUSAB, 40.0)
    private val pusab50 = loadFont(PUSAB, 50.0)

    init {
        initialize()
    }

    override fun initialize() {
        logger.info("Pushing help state.")
    }

    override fun update() {
        handleInput()
    }

    override fun draw(graphics: GraphicsContext) {
        graphics.drawImage(getImage("state-help"), 0.0, 0.0)
        graphics.fill = Color.BLACK

        graphics.textAlign = TextAlignment.CENTER
        graphics.font = pusab50
        graphics.fillText(labelHelp, 320.0, 120.0)

        graphics.font = pusab40
        graphics.textAlign = TextAlignment.LEFT
        graphics.fillText(labelsControls.get(0), 330.0, 210.0)
        graphics.fillText(labelsControls.get(1), 330.0, 280.0)
        graphics.fillText(labelsControls.get(2), 330.0, 355.0)
        graphics.fillText(labelsControls.get(3), 330.0, 460.0)

        graphics.font = pusab50
        graphics.fill = Color.RED
        graphics.textAlign = TextAlignment.CENTER
        graphics.fillText(labelExit, 320.0, 545.0)
        graphics.drawImage(getImage("arrow-right"), 180.0, 512.0)
    }

    override fun handleInput() {
        if (KeyManager.isPressed(KeyManager.K_ENTER)) {
            logger.info("Popping help state. Going back to previous state.")
            gameStateManager.popState()
        }
    }
}